provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -0,0 +1,73 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::collections::BTreeMap;
+
+use crate::transform::cldr::cldr_serde;
+use icu_provider::{
+    datagen::IterableDataProvider, DataError, DataLocale, DataPayload, DataProvider, DataRequest,
+    DataResponse,
+};
+use icu_unitsconversion::provider::{UnitsConstantsV1, UnitsConstantsV1Marker};
+use zerovec::ZeroMap;
+
+impl DataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
+    fn load(&self, _req: DataRequest) -> Result<DataResponse<UnitsConstantsV1Marker>, DataError> {
+        self.check_req::<UnitsConstantsV1Marker>(_req)?;
+
+        let _units_data: &cldr_serde::units::units_constants::Resource = self
+            .cldr()?
+            .core()
+            .read_and_parse("supplemental/units.json")?;
+        let mut constants_map = BTreeMap::<&str, &str>::new();
+
+        let constants = &_units_data.supplemental.unit_constants.constants;
+        for (key, constant) in constants {
+            constants_map.insert(key, &constant.value);
+        }
+
+        let result = UnitsConstantsV1 {
+            constants_map: ZeroMap::from_iter(constants_map.into_iter()),
+        };
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(result)),
+        })
+    }
+}
+
+impl IterableDataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(self
```

Comment:
Isn't this a singleton?

Reply:
yes

Reply:
Your code lists out all the locales in the `cldr-numbers-full` directory

Reply:
Probably a copy-paste error, you want to just return `Ok(vec![Default::default()])`

Reply:
done